lib/gis/testsuite/test_lrand48.py

```diff
@@ -0,0 +1,108 @@
+"""Test of gis library lrand48 PRNG thread-safety
+
+@author Maris Nartiss
+@author Gemini
+
+@copyright 2025 by the GRASS Development Team
+
+@license This program is free software under the GNU General Public License (>=v2).
+Read the file COPYING that comes with GRASS
+for details
+"""
+
+import ctypes
+import threading
+
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+
+from grass.lib.gis import G_srand48, G_lrand48
+
+
+class Lrand48ThreadSafetyTestCase(TestCase):
+    """Test case for lrand48 thread-safety and reproducibility."""
+
+    def test_thread_safety_and_reproducibility(self):
+        """Verify that multi-threaded execution produces the same set of
+        random numbers as single-threaded execution."""
+
+        seed = 1337
+        num_values = 10000
+        num_threads = 4
+        values_per_thread = num_values // num_threads
+
+        self.assertEqual(
+            num_values % num_threads,
+            0,
+            "Total number of values must be divisible by the number of threads.",
+        )
+
+        # --- Define ctypes function signatures ---
+        G_srand48.argtypes = [ctypes.c_long]
+        G_srand48.restype = None
+
+        G_lrand48.argtypes = []
+        G_lrand48.restype = ctypes.c_long
+
+        # --- 1. Single-threaded execution ---
```

Comment:
While often seen in generated code, I think our experience is that these ASCII comment graphic is hard to maintain and inherently inconsistent (thinking about old code we cleaned in GRASS).